src/ngraph/op/fused/normalize_l2.cpp

```diff
@@ -42,65 +42,66 @@ op::NormalizeL2::NormalizeL2(const Output<Node>& data,
-                          "Input tensor rank must be 2, 3 or 4 dimensional (actual input "
-                          "shape: ",
-                          data_shape,
-                          ").");
+                          axes_node->is_constant(),
+                          "doesn't support 'axes' input of other type than a Constant.");
 
-    NODE_VALIDATION_CHECK(this,
-                          static_cast<size_t>(axes_pshape.rank()) == 1,
-                          "Input axes must have rank equals 1 (axes shape: ",
-                          axes_pshape,
-                          ").");
+    if (axes_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<size_t>(axes_pshape.rank()) == 1,
+                              "Input axes must have rank equals 1 (axes shape: ",
+                              axes_pshape,
+                              ").");
+
+        if (input_rank.is_static())
+        {
+            const auto reduction_axes = get_reduction_axes();
+            for (auto axis : reduction_axes)
+            {
+                NODE_VALIDATION_CHECK(this,
+                                      axis < size_t(input_rank),
+                                      "Reduction axis (",
+                                      axis,
+                                      ") is out of bounds ",
+                                      "(argument shape: ",
+                                      input_pshape,
+                                      ")");
+            }
+        }
+    }
 }
 
-NodeVector op::NormalizeL2::decompose_op() const
+AxisSet op::NormalizeL2::get_reduction_axes() const
 {
-    Output<Node> data{input_value(0)};
-    const Shape input_shape{data.get_shape()};
-
-    // Reshape to 4D tensor.
-    if (input_shape.size() != 4)
+    AxisSet axes;
+    auto axes_input_node = input_value(1).get_node_shared_ptr();
+    if (auto const_op = dynamic_pointer_cast<op::Constant>(axes_input_node))
     {
-        Shape data_shape(4 - input_shape.size(), 1);
-        copy(begin(input_shape), end(input_shape), back_inserter(data_shape));
-        data = builder::reshape(data, data_shape);
+        axes = const_op->get_axis_set_val();
     }
+    return axes;
+}
 
-    auto axes_node = input(1).get_source_output().get_node_shared_ptr();
-    NODE_VALIDATION_CHECK(this,
-                          axes_node->is_constant(),
-                          "doesn't support 'axes' input of other type than a Constant.");
+NodeVector op::NormalizeL2::decompose_op() const
+{
+    Output<Node> data{input_value(0)};
+    const Shape input_shape{data.get_shape()};
 
-    // Calculate norm over axes indicated by axes input param
-    auto axes_constant = dynamic_pointer_cast<op::Constant>(axes_node);
-    auto axes_vector = axes_constant->get_vector<size_t>();
-    AxisSet reduction_axes{axes_vector};
+    AxisSet reduction_axes = get_reduction_axes();
```

Comment:
Why you need axes set here and why not to pass 1 input (with axes) to the builder::l2_norm?  Otherwise it looks weird that if 1 input is not Constant then axes are empty set.

Reply:
- `l2_norm` takes as input `AxisSet` - we haven't access to `Output<Node>` data if it is not `Constant` in the `op` class. In order to support such case new version of op is need (NormalizeL2:v1) and new backends which can support it.
- In `pre_validate_and_infer_types` method (which is called always in constructor) is assert which provides that `axes_node->is_constant() == true`, so `get_reduction_axes` will be work correctly in any case

Reply:
Ok